### ISSUE TITLE
bug: apply_for_references validation applies to all transformations

### ISSUE THREAD

**Issue by janmeier** [GreenmaskIO/greenmask#235]
Given the following configuration:
```yaml
    - schema: public
      name: "account"
      transformers:
        - name: RandomInt
          apply_for_references: true
          params:
            min: 0
            max: 100
            column: "id1"
            engine: "hash"
        - name: RandomUuid
          params:
            column: "id2"
```

I want to apply the transformations for id1 to all references, but _not_ for id2. 
However, I get the following validation error:

```
ValidationWarning={"hash":"","meta":{"TransformerName":"RandomUuid"},"msg":"cannot apply transformer for references: engine parameter is not hash","severity":"error"}
```

So the validation for `apply_for_references` is being applied to `id2` as well.

**Comment by wwoytenko** [comment]
Thank you for reporting. I've just fixed it in #236. Added regression tests as well. 

I hope it resolved now

**Comment by wwoytenko** [comment]
I will publish a release with this patch tomorrow.

**Comment by wwoytenko** [comment]
Published in v0.2.2
